ingestion/src/metadata/workflow/ingestion.py

```diff
@@ -218,3 +224,22 @@ def validate(self):
             raise WorkflowExecutionError(
                 f"Profiler is not supported for the service connection: {self.config.source.serviceConnection}"
             )
+
+    def import_source_class(self) -> Type[Source]:
+        source_type = self.config.source.type.lower()
+        try:
+            return (
+                import_from_module(
+                    self.config.source.serviceConnection.root.config.sourcePythonClass
+                )
+                if source_type.startswith("custom")
+                else import_source_class(
+                    service_type=self.service_type, source_type=source_type
+                )
+            )
+        except DynamicImportException as e:
+            if source_type.startswith("custom"):
```

Comment:
nit - We use the `custom` string in a few places. Maybe a constant can help us ensure no typos